plugins/package-managers/maven/src/test/kotlin/utils/DependencyTreeParserTest.kt

```diff
@@ -0,0 +1,376 @@
+                            "3.14.0",
+                            "jar",
+                            "compile",
+                            ""
+                        ),
+                        DependencyTreeMojoNode(
+                            "org.apache.commons",
+                            "commons-text",
+                            "1.12.0",
+                            "bundle",
+                            "test",
+                            "foo"
+                        ),
+                        DependencyTreeMojoNode(
+                            "commons-logging",
+                            "commons-logging",
+                            "1.3.2",
+                            "jar",
+                            "compile",
+                            ""
+                        )
+                    )
+                }
+            }
+        }
+    }
+
+    "parseDependencyTree()" should {
+        "parse the dependency tree of a single project" {
+            val projectNode = DependencyTreeMojoNode(
+                "org.ossreviewtoolkit",
+                "ort",
+                "1.2.3-SNAPSHOT",
+                "pom",
+                "",
+                "",
+                listOf(
+                    DependencyTreeMojoNode(
+                        "org.apache.commons",
+                        "commons-configuration2",
+                        "2.11.0",
+                        "jar",
+                        "compile",
+                        "",
+                        listOf(
+                            DependencyTreeMojoNode(
+                                "org.apache.commons",
+                                "commons-lang3",
+                                "3.14.0",
+                                "jar",
+                                "compile",
+                                ""
+                            ),
+                            DependencyTreeMojoNode(
+                                "org.apache.commons",
+                                "commons-text",
+                                "1.12.0",
+                                "bundle",
+                                "test",
+                                "foo"
+                            ),
+                            DependencyTreeMojoNode(
+                                "commons-logging",
+                                "commons-logging",
+                                "1.3.2",
+                                "jar",
+                                "compile",
+                                ""
+                            )
+                        )
+                    )
+                )
+            )
+
+            val projectDependencies = parseDependencyTree(
+                inputStreamFor(projectNode),
+                listOf(createProject("ort"))
+            ).toList()
+
+            projectDependencies shouldHaveSize 1
+            compareNodes(projectNode, projectDependencies.first())
```

Comment:
Alternatively:

            projectDependencies.shouldBeSingleton {
                compareNodes(projectNode, it)
            }